src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -268,16 +269,19 @@ static void initTools()
     public static int getTotalArmorValue(EntityPlayer player)
     {
         int ret = 0;
-        for (int x = 0; x < player.inventory.armorInventory.size(); x++)
+        for (EntityEquipmentSlot slot : EntityEquipmentSlot.values())
         {
-            ItemStack stack = player.inventory.armorInventory.get(x);
-            if (stack.getItem() instanceof ISpecialArmor)
+            if(slot.getSlotType() == EntityEquipmentSlot.Type.ARMOR)
             {
-                ret += ((ISpecialArmor)stack.getItem()).getArmorDisplay(player, stack, x);
-            }
-            else if (stack.getItem() instanceof ItemArmor)
-            {
-                ret += ((ItemArmor)stack.getItem()).damageReduceAmount;
+               ItemStack stack = player.GetItemStackFromSlot(slot);
```

Comment:
This doesn't even compile, there is no such method in `EntityPlayer`.
Also, indentation.